include/dca/phys/dca_step/cluster_solver/shared_tools/accumulation/tp/g4_helper.cuh

```diff
@@ -122,6 +126,36 @@ inline __device__ unsigned int G4Helper::g4Index(int k1, int k2, int k_ex, int w
          sbdm_steps_[8] * w2 + sbdm_steps_[9] * w_ex;
 }
 
+inline __device__
+void G4Helper::getComputeRange(const int my_rank, const int mpi_size,
```

Comment:
We need to start having "device" unit tests but this will have to be covered by integration testing until I setup a framework for that.

Reply:
Thanks!

Reply:
Just a reminder to be careful in using 64bit integer expressions to avoid potential overflow with 32bit integer type. I think  'unsigned int' may be 32bits. Some intermediate arithmetic expressions may also need to be computed in 64bit. Note a complex*16 array with 2^32  entries take up  about 64 GBytes of memory.

Reply:
@efdazedo right, I forgot the device code also needs the overflow protection. I updated: https://github.com/STEllAR-GROUP/DCA/blob/distG4_pr/include/dca/phys/dca_step/cluster_solver/shared_tools/accumulation/tp/g4_helper.cuh#L130-L158

Reply:
I agree with @efdazedo but I think we need an issue filed and separate PR's to deal with the integral types used for indexes across a good many locations in the code. For this PR lets leave those you have added uint_64t since that is still an easy type to grep for in the code base.